src/schema/schema_stat.c

```diff
@@ -147,13 +147,28 @@ __wt_curstat_table_init(
 
     WT_ERR(__wt_scr_alloc(session, 0, &buf));
 
+    stats = &cst->u.dsrc_stats;
+
+    /*
+     * If gathering statistics for a simple table, retrieve the underlying file's statistics rather
+     * than going through the usual schema parsing flow. This avoids unnecessary overhead in the
+     * statistics gathering process.
+     *
+     * There will always be at least one column group for a table.
+     */
+    if (table->is_simple) {
+        WT_ERR(__wt_buf_fmt(session, buf, "statistics:%s", table->cgroups[0]->name));
```

Comment:
I think this could be simplified, you could do something like this:

```
uri = table->cgroups[0]->name;
if (!WT_PREFIX_SKIP(uri, "file:"))
            return (__wt_unexpected_object_type(session, uri, "file:"));
ret = __wt_curstat_file_init(session, uri, NULL, cfg, CST);
```

Rather than creating a `statistics:file...` URI, and then stripping the `statistics:` off again in the file sub-call.

Reply:
@agorrod I'm not sure if this approach will work - I think the uris can start with `colgroup:`, which wouldn't match with `file:`